UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -288,9 +274,30 @@ void OnDisable()
                 Academy.Instance.AgentAct -= AgentStep;
                 Academy.Instance.AgentForceReset -= _AgentReset;
             }
+            OnDisableHelper();
             m_Brain?.Dispose();
         }
 
+        void OnDisableHelper()
```

Comment:
Perhaps call this, `NotifyTrainerAgentDone` or something.  It wasn't very clear to me that the line 
```
m_Info.done = true;
``` 
was the whole reason this method exists.